src/OpenTelemetry/Metrics/CompositeMetricReader.cs

```diff
@@ -57,12 +60,117 @@ public CompositeMetricReader AddReader(MetricReader reader)
             };
             this.tail.Next = node;
             this.tail = node;
+            this.count++;
 
             return this;
         }
 
         public Enumerator GetEnumerator() => new Enumerator(this.head);
 
+        internal List<Metric> AddMetricsWithNoViews(Instrument instrument)
+        {
+            var metrics = new List<Metric>();
+            for (var cur = this.head; cur != null; cur = cur.Next)
+            {
+                var metric = cur.Value.AddMetricWithNoViews(instrument);
+                metrics.Add(metric);
+            }
+
+            return metrics;
+        }
+
+        internal void RecordSingleStreamLongMeasurements(List<Metric> metrics, long value, ReadOnlySpan<KeyValuePair<string, object>> tags)
+        {
+            Debug.Assert(metrics.Count == this.count, "The count of metrics to be updated for a CompositeReader must match the number of individual readers.");
+
+            int index = 0;
+            for (var cur = this.head; cur != null; cur = cur.Next)
+            {
+                if (metrics[index] != null)
+                {
+                    cur.Value.RecordSingleStreamLongMeasurement(metrics[index], value, tags);
+                }
+
+                index++;
+            }
+        }
+
+        internal void RecordSingleStreamDoubleMeasurements(List<Metric> metrics, double value, ReadOnlySpan<KeyValuePair<string, object>> tags)
+        {
+            Debug.Assert(metrics.Count == this.count, "The count of metrics to be updated for a CompositeReader must match the number of individual readers.");
+
+            int index = 0;
+            for (var cur = this.head; cur != null; cur = cur.Next)
+            {
+                if (metrics[index] != null)
+                {
+                    cur.Value.RecordSingleStreamDoubleMeasurement(metrics[index], value, tags);
+                }
+
+                index++;
+            }
+        }
+
+        internal List<List<Metric>> AddMetricsSuperListWithViews(Instrument instrument, List<MetricStreamConfiguration> metricStreamConfigs)
+        {
+            var metricsSuperList = new List<List<Metric>>();
+            for (var cur = this.head; cur != null; cur = cur.Next)
+            {
+                var metrics = cur.Value.AddMetricsListWithViews(instrument, metricStreamConfigs);
+                metricsSuperList.Add(metrics);
```

Comment:
only add if the metric.Count > 0

Reply:
Same as https://github.com/open-telemetry/opentelemetry-dotnet/pull/2596#discussion_r752685144